### PR TITLE
CSV upload

### PR DESCRIPTION
**Pull request by berggren**
Support CSV upload via REST upload API.

* Ref issue: #264 
* Reviewer: @someguyiknow

### CODE REVIEW

File: timesketch/lib/tasks.py [hunk 1]
```diff
@@ -26,6 +27,10 @@
     pass
 
 from timesketch import create_celery_app
+from timesketch.lib.datastores.elastic import ElasticSearchDataStore
```
Comment (someguyiknow) [inline timesketch/lib/tasks.py]: nit: In all the comments, Elasticsearch is used. But this lib is ElasticsearchDataStore.

File: timesketch/lib/datastores/elastic.py [hunk 2]
```diff
@@ -11,8 +11,9 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-"""ElasticSearch datastore."""
+"""Elasticsearch datastore."""
```
Comment (someguyiknow) [inline timesketch/lib/datastores/elastic.py]: nit: In all the comments, Elasticsearch is used. But this lib has class ElasticsearchDataStore.

**Comment by someguyiknow** [comment]
LGTM

File: timesketch/lib/datastores/elastic.py (hunk 2 above)
Comment (berggren) [inline timesketch/lib/datastores/elastic.py]: Good point, fixed now.

**Comment by berggren** [comment]
#195

**Comment by geekscrapy** [comment]
Any docs on how to actually upload a CSV via the API?

**Comment by someguyiknow** [comment]
Johan may be better situated to speak for documentation, but this may help:
https://github.com/log2timeline/dftimewolf/blob/master/dftimewolf/lib/timesketch_utils.py

**Comment by adamjnichols** [comment]
Hi there; we discuss this a bit in #522.  Tl;dr: the RESTful API is currently a means to provide data to the UI, but was obviously built with more in mind.  We’re hoping to expose something more widely in the future.

To Gary’s point, posting a CSV via REST is trivial, and something you should be able to work out from source until we get this sorted.  Have a look at [resources.py](https://github.com/google/timesketch/blob/master/timesketch/api/v1/resources.py)

**Comment by geekscrapy** [comment]
Thanks, I'll take a look

**Comment by berggren** [comment]
Also, there is the api-client package that you can use:

```$ pip install timesketch-api-client```

```
from timesketch_api_client.client import TimesketchApi

ts = TimesketchApi('https"//<timesketch server>', 'user', 'password')
sketch = ts.get_sketch(sketch_id=1)
sketch.upload('Name of timeline', '/path/to/file.cvs')
```

Code is here: https://github.com/google/timesketch/blob/master/api_client/python/timesketch_api_client/client.py

**Pull request merged**